From 3189e197981d0dd5f6eec6dabd9dedc6db7a21d3 Mon Sep 17 00:00:00 2001
From: Ken Brown <kbrown@cornell.edu>
Date: Sun, 26 Mar 2017 12:56:26 -0400
Subject: [PATCH 3/3] Fix "undefined symbols" errors on Cygwin

* lib/Makefile.am (libmuaux_la_LIBADD): Add the same libraries
needed for building libmailutils.
* pop3d/Makefile.am (pop3d_LDADD): Add libsockaddr.la.
* imap4d/Makefile.am (imap4d_LDADD): Add libsockaddr.la and
libstring.la.
* maidag/Makefile.am (maidag_LDADD): Add libsockaddr.la.
---
 imap4d/Makefile.am |  4 +++-
 lib/Makefile.am    | 28 ++++++++++++++++++++++++++++
 maidag/Makefile.am |  3 ++-
 pop3d/Makefile.am  |  3 ++-
 4 files changed, 35 insertions(+), 3 deletions(-)

diff --git a/imap4d/Makefile.am b/imap4d/Makefile.am
index e62c34d..451914b 100644
--- a/imap4d/Makefile.am
+++ b/imap4d/Makefile.am
@@ -79,7 +79,9 @@ imap4d_LDADD = \
  ${MU_LIB_AUTH}\
  @MU_AUTHLIBS@ \
  ${MU_LIB_MAILUTILS}\
- @SERV_AUTHLIBS@ @MU_COMMON_LIBRARIES@ @MU_TCPWRAP_LIBRARIES@
+ @SERV_AUTHLIBS@ @MU_COMMON_LIBRARIES@ @MU_TCPWRAP_LIBRARIES@\
+ ../libmailutils/sockaddr/libsockaddr.la\
+ ../libmailutils/string/libstring.la
 
 ## This kludge is necessary to correctly establish imap4d -> MU_AUTHLIBS
 ## dependencies. Automake stupidly refuses to include them.
diff --git a/lib/Makefile.am b/lib/Makefile.am
index 94e9593..ac8ff4e 100644
--- a/lib/Makefile.am
+++ b/lib/Makefile.am
@@ -36,6 +36,34 @@ libmuaux_la_SOURCES = \
  mu_umaxtostr.c\
  mu_umaxtostr.h
 libmuaux_la_LIBADD=gnu/libgnu.la
+
+## The following list is taken from ../libmailutils/Makefile.am
+libmuaux_la_LIBADD += \
+ @MU_COMMON_LIBRARIES@\
+ ../libmailutils/auth/libauth.la\
+ ../libmailutils/base/libbase.la\
+ ../libmailutils/address/libaddress.la\
+ ../libmailutils/sockaddr/libsockaddr.la\
+ ../libmailutils/cidr/libcidr.la\
+ ../libmailutils/cfg/libcfg.la\
+ ../libmailutils/cli/libcli.la\
+ ../libmailutils/datetime/libdatetime.la\
+ ../libmailutils/diag/libdiag.la\
+ ../libmailutils/filter/libfilter.la\
+ ../libmailutils/imapio/libimapio.la\
+ ../libmailutils/list/liblist.la\
+ ../libmailutils/mailbox/libmailbox.la\
+ ../libmailutils/mailer/libmailer.la\
+ ../libmailutils/mime/libmime.la\
+ ../libmailutils/msgset/libmsgset.la\
+ ../libmailutils/opt/libopt.la\
+ ../libmailutils/property/libproperty.la\
+ ../libmailutils/server/libserver.la\
+ ../libmailutils/string/libstring.la\
+ ../libmailutils/stream/libstream.la\
+ ../libmailutils/stdstream/libstdstream.la\
+ ../libmailutils/url/liburl.la
+
 libmuaux_la_LDFLAGS = -no-undefined -version-info @VI_CURRENT@:@VI_REVISION@:@VI_AGE@
 
 libmutcpwrap_a_SOURCES = tcpwrap.c
diff --git a/maidag/Makefile.am b/maidag/Makefile.am
index 1a4940a..8cbe18b 100644
--- a/maidag/Makefile.am
+++ b/maidag/Makefile.am
@@ -53,7 +53,8 @@ maidag_LDADD = \
  @GUILE_LIBS@\
  @PYTHON_LIBS@\
  @DBMLIBS@\
- @MU_TCPWRAP_LIBRARIES@
+ @MU_TCPWRAP_LIBRARIES@\
+ ../libmailutils/sockaddr/libsockaddr.la
 
 AM_CPPFLAGS = -I${top_srcdir} @MU_APP_COMMON_INCLUDES@ @GUILE_INCLUDES@ \
  @PYTHON_INCLUDES@
diff --git a/pop3d/Makefile.am b/pop3d/Makefile.am
index ba21c9a..e87c0ba 100644
--- a/pop3d/Makefile.am
+++ b/pop3d/Makefile.am
@@ -61,7 +61,8 @@ pop3d_LDADD = \
  @MU_COMMON_LIBRARIES@\
  ${LIBMU_DBM}\
  @DBMLIBS@\
- @MU_TCPWRAP_LIBRARIES@
+ @MU_TCPWRAP_LIBRARIES@\
+ ../libmailutils/sockaddr/libsockaddr.la
 
 popauth_SOURCES = popauth.c
 popauth_LDADD = \
-- 
2.8.3

